### DISCUSSION TITLE
TimeInput needs improvements (12-hour / 24-hour, AM / PM, custom date ranges)

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [mantinedev/mantine#6399] [Feature requests]
We know for `<TimeInput />` component you used `input[type="time"]` but to be honest it's hard to change things using it, and I see a lot of frustrated people by this decision (https://github.com/mantinedev/mantine/issues/4213#issuecomment-1540734888) which prevents from easy management.

For example, our customers need:
- 12-hour / 24-hour switch
- AM / PM support
- Custom date "steps" (e.g. 30-min ranges) -> https://github.com/mantinedev/mantine/issues/5943
- Native blue color is quite random, inconsistent with our platform
![Screenshot 2024-06-16 at 22 20 13](https://github.com/mantinedev/mantine/assets/22228431/9511227c-45af-4f6b-8352-20de597314a3)

I know it's a lot of work to rebuild it without `input[type="time"]` but it's super hard to change things using it.
We could write our own wrapper using `<Select />` but if this feature could address those problems above it would be super appreciated.

**Comment by atumas-bananamilk** [comment]
Something like this, called "step" would solve "custom date ranges":
https://hilla.dev/docs/react/components/time-picker#step

**Comment by atumas-bananamilk** [comment]
👋

**Comment by rtivital** [comment]
It will be added in 8.0 next year

**Reply by Deepanshu-T2b** [reply]
It's still not added in 8.0 version, waiting for this change

**Reply by rtivital** [reply]
It is added as TimePicker component

**Reply by Deepanshu-T2b** [reply]
Thanks for the update! I appreciate you letting me know that it has been added as a TimePicker component.

**Comment by martijnheinisarchitechts** [comment]
Not having the option to use 12-hr format or 24-hr format is kind of a deal breaker at the moment for our projects using Mantine. So yeah, happy with the announcement about the adding 👍
